toolchain/base/value_store.h

```diff
@@ -86,25 +110,15 @@ struct IdTag {
       return {-1, tagged_index};
     }
     auto index_mask = llvm::maskTrailingOnes<uint32_t>(location);
-    auto ir_id = (llvm::reverseBits(tagged_index & ~index_mask) >> 2) - 1;
+    auto check_ir_id = (llvm::reverseBits(tagged_index & ~index_mask) >> 2) - 1;
     auto index = tagged_index & index_mask;
-    return {ir_id, index};
-  }
-  auto Remove(int32_t tagged_index) const -> int32_t {
-    if ((llvm::reverseBits(2) & tagged_index) == 0) {
-      CARBON_DCHECK(tagged_index < initial_reserved_ids_,
-                    "This untagged index is outside the initial reserved ids "
-                    "and should have been tagged.");
-      return tagged_index;
-    }
-    auto index = tagged_index ^ id_tag_;
-    CARBON_DCHECK(index >= initial_reserved_ids_,
-                  "When removing tagging bits, found an index that "
-                  "shouldn't've been tagged in the first place.");
-    return index;
+    return {.check_ir_id = static_cast<int32_t>(check_ir_id),
+            .index = static_cast<int32_t>(index)};
   }
-  int32_t id_tag_;
-  int32_t initial_reserved_ids_;
+
+ private:
+  int32_t id_tag_ = 0;
+  int32_t initial_reserved_ids_ = std::numeric_limits<int32_t>::max();
```

Comment:
Why did you switch away from the prior setup? It seemed safer, since there are only two constructors and both must set different values.